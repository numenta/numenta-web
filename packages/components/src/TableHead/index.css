/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.tableHeader {
  /* */
}
