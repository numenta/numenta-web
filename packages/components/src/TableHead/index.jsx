// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


/**
 *
 */
const TableHeader = ({border, children}) => {
  const childrenWithProps = React.Children.map(children, (child) =>
    React.cloneElement(child, {border})
  )

  return (
    <thead className={styles.tableHeader}>
      {childrenWithProps}
    </thead>
  )
}

TableHeader.propTypes = {
  border: React.PropTypes.bool,
  children: React.PropTypes.node.isRequired,
}

export default TableHeader
