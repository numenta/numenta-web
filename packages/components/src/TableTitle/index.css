/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.border {
  composes: b--black-10 from 'tachyons';
  composes: ba from 'tachyons';
}

.tableTitle {
  composes: bg-near-white from 'tachyons';
  composes: pa2 from 'tachyons';
  composes: v-top from 'tachyons';
}
