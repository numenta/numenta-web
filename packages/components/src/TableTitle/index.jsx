// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


/**
 *
 */
const TableTitle = ({border, children, colSpan}) => {
  const classes = [styles.tableTitle]

  if (border) classes.push(styles.border)

  return (
    <th className={classNames(...classes)} colSpan={colSpan}>
      {children}
    </th>
  )
}

TableTitle.propTypes = {
  border: React.PropTypes.bool,
  children: React.PropTypes.node,
  colSpan: React.PropTypes.number.isRequired,
}

TableTitle.defaultProps = {
  colSpan: 1,
}

export default TableTitle
