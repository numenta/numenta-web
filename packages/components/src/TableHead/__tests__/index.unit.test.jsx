import React from 'react'
import renderer from 'react-test-renderer'

import TableHead from '../../TableHead'
import TableRow from '../../TableRow'
import TableTitle from '../../TableTitle'


describe('TableHead React component', () => {

  it('Renders Bordered correctly', () => {
    const component = renderer.create(
      <TableHead border={true}>
        <TableRow>
          <TableTitle>
            Table Title 1
          </TableTitle>
          <TableTitle>
            Table Title 2
          </TableTitle>
        </TableRow>
      </TableHead>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Un-bordered correctly', () => {
    const component = renderer.create(
      <TableHead border={false}>
        <TableRow>
          <TableTitle>
            Table Title 1
          </TableTitle>
          <TableTitle>
            Table Title 2
          </TableTitle>
        </TableRow>
      </TableHead>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
