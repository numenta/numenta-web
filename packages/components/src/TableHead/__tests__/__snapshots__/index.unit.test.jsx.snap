exports[`TableHead React component Renders Bordered correctly 1`] = `
<thead
  className="tableHeader">
  <tr
    className="tableRow">
    <th
      className="tableTitle border"
      colSpan={1}>
      Table Title 1
    </th>
    <th
      className="tableTitle border"
      colSpan={1}>
      Table Title 2
    </th>
  </tr>
</thead>
`;

exports[`TableHead React component Renders Un-bordered correctly 1`] = `
<thead
  className="tableHeader">
  <tr
    className="tableRow">
    <th
      className="tableTitle"
      colSpan={1}>
      Table Title 1
    </th>
    <th
      className="tableTitle"
      colSpan={1}>
      Table Title 2
    </th>
  </tr>
</thead>
`;
